deploy-agent/deployd/client/serverless_client.py

```diff
@@ -26,7 +26,7 @@
 
 log: logging.Logger = logging.getLogger(__name__)
 
-_DEPLOY_STAGE_TRANSITIONS = dict([(i, i+1) for i in range(DeployStage.PRE_DOWNLOAD, DeployStage.SERVING_BUILD)])
+_DEPLOY_STAGE_TRANSITIONS = { DeployStage(i).value : DeployStage(i+1).value for i in range(DeployStage.PRE_DOWNLOAD.value, DeployStage.SERVING_BUILD.value) }
```

Comment:
It seems a bit brittle to rely on the enum order, just a nit. I think it's alright how it is.

Reply:
Created a ticket for this